libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationViewModel.java

```diff
@@ -443,11 +452,19 @@ private void sendEventArrival(RouteProgress routeProgress, Milestone milestone)
     }
   }
 
-  private void handleOffRouteEvent(Point newOrigin) {
+  private void handleOffRouteEvent(Location location) {
+    Point newOrigin = Point.fromLngLat(location.getLongitude(), location.getLatitude());
     if (navigationViewEventDispatcher != null && navigationViewEventDispatcher.allowRerouteFrom(newOrigin)) {
       navigationViewEventDispatcher.onOffRoute(newOrigin);
       OffRouteEvent event = new OffRouteEvent(newOrigin, routeProgress);
-      navigationViewRouteEngine.fetchRouteFromOffRouteEvent(event);
+
+      if (isOffline) {
+        ArrayList<Point> locations = viewRouteFetcher.calculateRemainingCoordinates(event);
+        navigation.findOfflineRouteFor(location, locations);
```

Comment:
It looks like we are finding the route here, but not updating the `DirectionsRoute` with `updateRoute(DirectionsRoute route)` (we do this when we get a route back with `ViewRouteFetcher`.  So the route found will never propagate here

Reply:
@danesfeder thanks! I was worried I was missing something like that